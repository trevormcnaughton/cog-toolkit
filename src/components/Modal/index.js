import React, { PureComponent } from 'react'
import theme from '../../util/theme'
import styled from 'styled-components'

const StyledModal = styled.section`
  display: flex;
  position: fixed;
  z-index: 9990;
  left: 0;
  right: 0;
  bottom: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.64);
  transition: opacity 200ms;
  will-change: opacity;
`

const Frame = styled.section`
  position: relative;
  margin: auto;
  width: 32rem;
  background: #fff;
  box-shadow: 0 16px 32px 0 rgba(0, 0, 0, 0.24);
  border-radius: 4px;
  overflow: hidden;
  transition: transform 400ms 200ms cubic-bezier(0.175, 0.665, 0.320, 1),
    opacity 400ms 200ms;
  will-change: transform;
`

const Header = styled.header`
  display: flex;
  height: 3rem;
  align-items: center;
  justify-content: space-between;
  padding: 0.25rem 1rem;
  border-bottom: 1px solid ${theme.color.greyLight};
`

const Title = styled.h1`
  font-size: 1rem;
  font-weight: 900;
`

const CloseButton = styled.section`
  font-size: 1.5rem;
  cursor: pointer;
`

const Body = styled.section`
  position: relative;
  padding: 1.5rem;
`

export default class Modal extends PureComponent {
  render() {
    const {
      children,
      className,
      onDismissal,
      title,
      show,
      large,
      small,
    } = this.props

    return show
      ? <StyledModal>
          <Frame>
            {title &&
              <Header>
                <Title>
                  {title}
                </Title>
                <CloseButton onClick={onDismissal}>&times;</CloseButton>
              </Header>}
            {children}
          </Frame>
        </StyledModal>
      : null
  }

  static Body = Body
}
