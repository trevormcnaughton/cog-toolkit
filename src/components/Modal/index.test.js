import React from 'react'
import renderer from 'react-test-renderer'

import Modal from './index'

const noop = () => {}

// TODO: Add interactivity tests

describe('Modal', () => {
  describe('snapshots', () => {
    it('renders a hidden modal', () => {
      const tree = renderer.create(
        <Modal show={false}>
          <p>Some modal</p>
        </Modal>
      )

      expect(tree).toMatchSnapshot()
    })
    it('renders a visible modal', () => {
      const tree = renderer.create(
        <Modal show={true}>
          <p>Some modal</p>
        </Modal>
      )

      expect(tree).toMatchSnapshot()
    })
    it('renders a modal with header', () => {
      const tree = renderer.create(
        <Modal
          title="My Modal"
          show={true}
          onDismissal={noop}>
          <Modal.Body>Modal Stuff!</Modal.Body>
        </Modal>
      )

      expect(tree).toMatchSnapshot()
    })
  })
})
