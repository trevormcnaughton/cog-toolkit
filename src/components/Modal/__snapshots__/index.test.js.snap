// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal snapshots renders a hidden modal 1`] = `null`;

exports[`Modal snapshots renders a modal with header 1`] = `
<section
  className="sc-bdVaJa iEETSP"
>
  <section
    className="sc-bwzfXH jMroiM"
  >
    <header
      className="sc-htpNat biZOlL"
    >
      <h1
        className="sc-bxivhb fBkLCD"
      >
        My Modal
      </h1>
      <section
        className="sc-ifAKCX jMvEhG"
        onClick={[Function]}
      >
        Ã—
      </section>
    </header>
    <section
      className="sc-EHOje eizxor"
    >
      Modal Stuff!
    </section>
  </section>
</section>
`;

exports[`Modal snapshots renders a visible modal 1`] = `
<section
  className="sc-bdVaJa iEETSP"
>
  <section
    className="sc-bwzfXH jMroiM"
  >
    <p>
      Some modal
    </p>
  </section>
</section>
`;
